src/node.cc

```diff
@@ -1148,7 +1149,15 @@ void TearDownOncePerProcess() {
 }
 
 int Start(int argc, char** argv) {
-  InitializationResult result = InitializeOncePerProcess(argc, argv);
+  node::single_binary::NewArgs* newArgs =
+     node::single_binary::checkForSingleBinary(argc, argv);
+
+  InitializationResult result;
+  if (!newArgs->singleBinary) {
+    result = InitializeOncePerProcess(argc, argv);
+  } else {
+    result = InitializeOncePerProcess(newArgs->argc, newArgs->argv);
```

Comment:
Injecting the JS code string in this way will force the code to be run in a REPL like environment because of the `-e` option. I don't think we should do something like that because it would introduce globals like `fs`, `net`, etc. which one might not expect. Passing the JS string as a second argument to https://github.com/nodejs/node/blob/a346572b952e7942078af4105d34e049d28d3ea2/src/node_main_instance.cc#L138 might fix that.